Import Library

In [1]:
import numpy as np
import pandas as pd

In [2]:
#Get the csv file
data=pd.read_csv("msg_data.csv")
data.head()

Category                                            Message
0      ham  Go until jurong point, crazy.. Available only ...
1      ham                      Ok lar... Joking wif u oni...
2     spam  Free entry in 2 a wkly comp to win FA Cup fina...
3      ham  U dun say so early hor... U c already then say...
4      ham  Nah I don't think he goes to usf, he lives aro...

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5572 entries, 0 to 5571
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Category  5572 non-null   object
 1   Message   5572 non-null   object
dtypes: object(2)
memory usage: 87.2+ KB


In [4]:
#Checking null Values in dataset
data.isnull().sum()

Category    0
Message     0
dtype: int64

In [5]:
data

Category                                            Message
0         ham  Go until jurong point, crazy.. Available only ...
1         ham                      Ok lar... Joking wif u oni...
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...
3         ham  U dun say so early hor... U c already then say...
4         ham  Nah I don't think he goes to usf, he lives aro...
...       ...                                                ...
5567     spam  This is the 2nd time we have tried 2 contact u...
5568      ham               Will ü b going to esplanade fr home?
5569      ham  Pity, * was in mood for that. So...any other s...
5570      ham  The guy did some bitching but I acted like i'd...
5571      ham                         Rofl. Its true to its name

[5572 rows x 2 columns]

In [6]:
msg_data=data.where((pd.notnull(data)),'')

In [7]:
msg_data

Category                                            Message
0         ham  Go until jurong point, crazy.. Available only ...
1         ham                      Ok lar... Joking wif u oni...
2        spam  Free entry in 2 a wkly comp to win FA Cup fina...
3         ham  U dun say so early hor... U c already then say...
4         ham  Nah I don't think he goes to usf, he lives aro...
...       ...                                                ...
5567     spam  This is the 2nd time we have tried 2 contact u...
5568      ham               Will ü b going to esplanade fr home?
5569      ham  Pity, * was in mood for that. So...any other s...
5570      ham  The guy did some bitching but I acted like i'd...
5571      ham                         Rofl. Its true to its name

[5572 rows x 2 columns]

In [8]:
msg_data.shape

(5572, 2)

In [9]:
msg_data.head()

Category                                            Message
0      ham  Go until jurong point, crazy.. Available only ...
1      ham                      Ok lar... Joking wif u oni...
2     spam  Free entry in 2 a wkly comp to win FA Cup fina...
3      ham  U dun say so early hor... U c already then say...
4      ham  Nah I don't think he goes to usf, he lives aro...

In [10]:
#Label ham mail as 1, Spam as 0.
y=msg_data['Category'].map({'ham':1,'spam':0})

In [11]:
y

0       1
1       1
2       0
3       1
4       1
       ..
5567    0
5568    1
5569    1
5570    1
5571    1
Name: Category, Length: 5572, dtype: int64

In [12]:
#Seprate X and y data.....
X=msg_data['Message']

In [13]:
X

0       Go until jurong point, crazy.. Available only ...
1                           Ok lar... Joking wif u oni...
2       Free entry in 2 a wkly comp to win FA Cup fina...
3       U dun say so early hor... U c already then say...
4       Nah I don't think he goes to usf, he lives aro...
                              ...                        
5567    This is the 2nd time we have tried 2 contact u...
5568                 Will ü b going to esplanade fr home?
5569    Pity, * was in mood for that. So...any other s...
5570    The guy did some bitching but I acted like i'd...
5571                           Rofl. Its true to its name
Name: Message, Length: 5572, dtype: object

In [14]:
#Import train_test_split And Devide data into train and test data
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=3)

In [15]:
#Import TfidfVectorizer to convert text data into number
from sklearn.feature_extraction.text import TfidfVectorizer

In [16]:
#fit and transform train and test data
feature_extraction=TfidfVectorizer(min_df=1,stop_words='english',lowercase=True)
X_train_feature=feature_extraction.fit_transform(X_train)
X_test_feature=feature_extraction.transform(X_test)

In [ ]:
y_train.info()

In [17]:
#Select logistic Regression and import accuracy_score
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

In [18]:
# fit data into model
model=LogisticRegression()
model.fit(X_train_feature,y_train)

LogisticRegression()

In [19]:
#finding prediction on train data and then Check Accuracy of train data predction
pred_on_traindata=model.predict(X_train_feature)
accuracy_on_traindata=accuracy_score(y_train,pred_on_traindata)

In [20]:
accuracy_on_traindata

0.9661538461538461

In [21]:
#finding prediction on test data and then Check Accuracy of test data predction
pred_on_testdata=model.predict(X_test_feature)
accuracy_on_testdata=accuracy_score(y_test,pred_on_testdata)

In [22]:
accuracy_on_testdata

0.9647129186602871

We Get good accuracy of our model
train_accuracy=96.6%
test_accuraacy=96.4%

Lets do some prediction.........

In [23]:
demo_msg=["You have exhausted your daily high speed internet quota of 2 GB on your Jio Number 7976135187 as of 02-Nov-23 19:51 Hrs.Your daily high speed internet quota will be restored by 03-Nov-23 00:47 tonight.Now you can continue enjoying high speed data by taking an EMERGENCY DATA VOUCHER and pay later. You can borrow 2 GB in form of data vouchers worth Rs. 25. "]
input_feature=feature_extraction.transform(demo_msg)

In [24]:
pred=model.predict(input_feature)

if(pred[0]==1):
    print(" This is Ham mail")
else:
    print(" This is Spam Mail")

 This is Ham mail
